modules/adspendBidAdapter.js

```diff
@@ -0,0 +1,150 @@
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'adspend';
+const BID_URL = '//rtb.com.ru/headerbidding-bid';
+const SYNC_URL = '//rtb.com.ru/headerbidding-sync?uid={UUID}';
+const COOKIE_NAME = 'hb-adspend-id';
+const UUID_LEN = 36;
+const COOKIE_EXPIRE = 600;
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['adspend'],
+  supportedMediaTypes: [BANNER],
+
+  onBidWon: function(winObj) {
+    const requestId = winObj.requestId;
+    const cpm = winObj.cpm;
+    const event = getWinEvent(requestId).replace(/\$\{AUCTION_PRICE\}/, cpm);
+
+    fetch(event);
+  },
+
+  isBidRequestValid: function(bid) {
+    const conf = config.getConfig();
+    return !!(conf.currency &&
+      conf.currency.adServerCurrency &&
+      bid.crumbs.pubcid &&
```

Comment:
The documentation should probably specify somewhere that the pubcid module is required

Reply:
Do you mean `adspendBidAdapter.md` file?